force_bdss/data_sources/data_source_utilities.py

```diff
@@ -188,3 +212,53 @@ def merge_lists_with_check(new_list, old_list, attributes):
                 new_element, old_element, attr, attributes,
                 ignore_default=True
             )
+
+
+def retain_list(new_list, old_list, attributes):
+    """Cross checks all elements in new_list against old_list
+    to identify those that are similar enough to be considered
+    as relating to the same variable.
+    Returns a list containing the old instances of elements
+    that match, and any elements in new_list that are not in old_list
+
+    Parameters
+    ----------
+    new_list: list
+        List of new elements
+    old_list: list
+        List of old elements to be retained
+    attributes: str or list of str, optional
+        An attribute or list of attributes to check equivalence on both
+        source and target objects
+
+    Returns
+    -------
+    retained_list: list
+        List containing all elements to be retained
+    """
+    retained_list = []
+    old_list = copy.copy(old_list)
+
+    for new_element in new_list:
+        retained = False
+        for old_element in old_list:
```

Comment:
This refers to my previous comment in [here](https://github.com/force-h2020/force-bdss/pull/230/files#r345851967)

A possible way to make this more robust (this is not a working solution but just a sketch) is:
```
retained_list = []
    old_list = copy.copy(old_list)

    min_len = min(len(new_list), len(old_list))
    merge_lists(new_list[:min_len], old_list[:min_len], attributes)
    retained_list = old_list[:min_len]

    for new_element in new_list[min_len:]:
        retained = False
        for old_element in old_list[min_len:]:
```